src/shared/coredump-util.h

```diff
@@ -0,0 +1,27 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+#pragma once
+
+#include "macro.h"
+
+typedef enum CoredumpFilter {
+        COREDUMP_FILTER_PRIVATE_ANONYMOUS   = 0,
+        COREDUMP_FILTER_SHARED_ANONYMOUS    = 1,
+        COREDUMP_FILTER_PRIVATE_FILE_BACKED = 2,
+        COREDUMP_FILTER_SHARED_FILE_BACKED  = 3,
+        COREDUMP_FILTER_ELF_HEADERS         = 4,
+        COREDUMP_FILTER_PRIVATE_HUGE        = 5,
+        COREDUMP_FILTER_SHARED_HUGE         = 6,
+        COREDUMP_FILTER_PRIVATE_DAX         = 7,
+        COREDUMP_FILTER_SHARED_DAX          = 8,
+        _COREDUMP_FILTER_MAX                = 9,
+        _COREDUMP_FILTER_INVALID            = -1,
+} CoredumpFilter;
+
+#define COREDUMP_FILTER_MASK_DEFAULT (1u << COREDUMP_FILTER_PRIVATE_ANONYMOUS | \
+                                      1u << COREDUMP_FILTER_SHARED_ANONYMOUS | \
+                                      1u << COREDUMP_FILTER_ELF_HEADERS | \
+                                      1u << COREDUMP_FILTER_PRIVATE_HUGE)
```

Comment:
i'd probably define COREDUMP_FILTER_TO_MASK() which does the shifting. Just to make this more readable

Reply:
It's just a shift... I think the mask would make this painfully verbose and not really clearer. (And there's two places where this would be used, so I don't think it's worth it.)